### PR TITLE
Added format and test cases for the 24 hrs - decimal time parsing

### LINKED ISSUE #571: dateparser.parse returns None for any datetime using 24hr format and time period when hour is > 12
Whils't parsing some data I got an error when processing `Sep 28, 2016, 20:13 pm`, I believe this is due to a conflict between 24 hour time format and a time period being supplied.
```
>>> print(dateparser.parse('Sep 28, 2016, 12:13 PM'))
2016-09-28 12:13:00
>>> print(dateparser.parse('Sep 28, 2016, 11:13 PM'))
2016-09-28 23:13:00
>>> print(dateparser.parse('Sep 28, 2016, 13:13 PM'))
None
>>> print(dateparser.parse('Sep 28, 2016, 13:13 AM'))
None
>>> 
```
I know this isn't technically valid since the period is not needed with 24hr format. Instead of returning `None` I believe that it would be useful to instead just ignore the given period and fallback on the given 24hr format.

i.e.:
`Sep 28, 2016, 13:13 AM` -> `Sep 28, 2016, 13:13`
`Sep 28, 2016, 13:13 PM` -> `Sep 28, 2016, 13:13`

### PR DESCRIPTION
**Pull request by Avilad0**
Fixes #571, closes #641

### CODE REVIEW

**Comment by Gallaecio** [comment]
Could you share references to real life scenarios where these formats are used? Also, I noticed you added 4 formats but only 2 tests.

**Comment by Avilad0** [comment]
@Gallaecio  sorry for the delay, i was occupied in some other work. I have added 2 more test cases which do not have am/pm reference which are the remaining 2 test cases. Please refer it.

**Comment by Gallaecio** [comment]
Could you share URLs to a couple of online resources where these formats are used?

**Comment by Avilad0** [comment]
https://strftime.org/   -  this link defines it purpose 

it basically adds the functionality to accept time without mentioning in specific format HH:MM:SS (double digits).    eg. 1:1:1 instead of writing 01:01:01

**Comment by Gallaecio** [comment]
The line switches at `dateparser/parser.py` seem wrong, an accident?

**Comment by Avilad0** [comment]
Yes, sorry it was an an accident.

File: dateparser/parser.py [hunk 1]
```diff
@@ -81,6 +81,10 @@ class _time_parser(object):
         '%I %p',
         '%H:%M:%S.%f',
         '%I:%M:%S.%f %p',
+        '%-H:%-M',
+        '%-H:%-M:%-S',
+        '%-H:%-M %p',
+        '%-H:%-M:%-S %p'
```
Comment (Gallaecio) [inline dateparser/parser.py:87]: Woudn’t the fix for https://github.com/scrapinghub/dateparser/issues/571 be to simply add `'%H:%M %p'`? I’m worried that we are opening the door here for wrong interpretations, to parse as dates stuff that is not a date. None of the input strings from the test look to me like real-life date strings. I could understand the hours part having a single digit, and if that requires `%-H` then `%-H` should be OK, but `%-M` and `%-S` seem wrong to me, and I would rather keep them out of the default list.

Comment (Avilad0) [reply dateparser/parser.py:87]: Yeah, That may be the case then we can simply add the single digit format for hours only as in '%-H:%M' etc.

Comment (noviluni) [reply dateparser/parser.py:87]: Hi @Gallaecio 
I needed some time to realize it, but it doesn't make any sense to add `'%-H:%M %p'` as is equivalent to `'%I:%M %p'` for values where hour < 10 and equivalent to `'%H:%M %p'` for values where hour >= 10

Anyway, I have just merged this: https://github.com/scrapinghub/dateparser/pull/634

**Pull request closed without merging**